pkg/kubelet/util/csr/csr.go

```diff
@@ -50,30 +55,46 @@ func RequestNodeCertificate(client certificatesclient.CertificateSigningRequestI
 	if err != nil {
 		return nil, fmt.Errorf("unable to generate certificate request: %v", err)
 	}
-	return RequestCertificate(client, csrData, []certificates.KeyUsage{
+	hash := sha512.New512_256()
+	hash.Write(privateKeyData)
+	hash.Write([]byte(subject.CommonName))
+	for _, org := range subject.Organization {
+		hash.Write([]byte(org))
+	}
+	name := "node-csr-" + base64.StdEncoding.EncodeToString(hash.Sum(nil))
+	return requestCertificate(client, name, csrData, []certificates.KeyUsage{
 		certificates.UsageDigitalSignature,
 		certificates.UsageKeyEncipherment,
 		certificates.UsageClientAuth,
 	})
 }
 
-// RequestCertificate will create a certificate signing request using the PEM
+// requestCertificate will create a certificate signing request using the PEM
```

Comment:
Update to include the purpose of the name argument, that existing requests matching the name will be re-used.